homeworks/kirkovvalentiniván/06_testing/test_bankaccount.py

```diff
@@ -0,0 +1,52 @@
+import pytest
+from bankaccount import BankAccount
+from unittest.mock import patch
+
+@pytest.fixture
+def bankaccount_1():
+    return BankAccount("David", 200)
+
+@pytest.fixture
+def bankaccount_2():
+    return BankAccount("Emma", 550)
+
+@pytest.fixture
+def invalid_bankaccount():
+    return BankAccount("Invalid Account", -100)
+
+@pytest.mark.parametrize("deposit_amount, expected_balance, expected_exception", [
+    (30, 230, None), #normal case
+    (0, 200, ValueError), #Error: adding zero to the actual balance 
+    (-30, 200, ValueError) #Error: adding negative number to the actual balance
+])
+
+def test_deposit(bankaccount_1, deposit_amount, expected_balance, expected_exception):
+    if expected_exception:
+        with pytest.raises(expected_exception):
+            bankaccount_1.deposit(deposit_amount)
+    else:
+        bankaccount_1.deposit(deposit_amount)
+        assert bankaccount_1.get_balance() == expected_balance
+
+def test_withdraw(bankaccount_1):
+    bankaccount_1.withdraw(50)
+    assert bankaccount_1.get_balance() == 150
+
+    with pytest.raises(ValueError):
+        bankaccount_1.withdraw(2000)
+
+    with pytest.raises(ValueError):
+        bankaccount_1.withdraw(0)
+
+def test_transfer(bankaccount_1, bankaccount_2):
```

Comment:
itt egy kicsit most összemosódtak a dolgok. 
megpróbálsz küldeni egy nem bankaccount object-nek 50 dollárt:

`bankaccount_1.transfer(50, "not a bank account")`

ettől ugye azt várjuk hogy TypeError-t dobjon, ez oké is. Viszont ezután megnézed hogy a bankaccount_1 balance-a 150 e. Nem lesz 150 a balance, mert előtte egy nem létező accountnak próbáltál meg a 200-ból 50-et elküldeni, így maradt 200 a balance.
ezért kapod ezt a hibát:
`FAILED homeworks/kirkovvalentiniván/06_testing/test_bankaccount.py::test_transfer - assert 200 == 150`

Szedd szét a két tesztet: legyen egy test_transfer_to_non_bankaccount_object() ahol csak ezt csinálod:
```
    with pytest.raises(TypeError):
        bankaccount_1.transfer(50, "not a bank account")
```

mivel ebben a tesztben csak arra vagyunk kiváncsiak hogy "ugye nem tudunk nem bankaccount objectnek pénzt küldeni?" 

aztán legyen egy test_successful_transfer() teszted ahol pedig küldesz a bankaccount1-től 50 dollárt a bankaccount2-nek, és utána megvizsgálod az assert-tel mindkettőnek a balance-át. Így működni fog.